packages/flame/lib/src/game/game_widget/game_widget.dart

```diff
@@ -23,11 +23,16 @@ typedef OverlayWidgetBuilder<T extends Game> = Widget Function(
   T game,
 );
 
+typedef GameCreate<T extends Game> = T Function();
+
 /// A [StatefulWidget] that is in charge of attaching a [Game] instance into the
 /// Flutter tree.
 class GameWidget<T extends Game> extends StatefulWidget {
-  /// The game instance in which this widget will render
-  final T game;
+  /// The game instance in which this widget will render.
+  final T? game;
```

Comment:
If this property shouldn't be accessed by the user, then it should probably be made private.

Also, the docstring on this property is no longer accurate.

Reply:
Usually, widget properties are not accessed by the user. Also, as final, I see no harm on letting the user see it accessing it.

Reply:
The harm in letting the user see it is that it makes API changes more difficult. If this property was private before, then this PR wouldn't have been breaking. By leaving this property public we run risk of having more breaking changes in the future.

Reply:
Another potential source of concern is that it might seem to the user that this property gives access to the `game` instance that runs inside the widget (even if the widget is .controlled), which would not be the case.

Reply:
Flutter widgets usually let the user access the fields, private attributes on widgets (which aren't internal by nature) are rare AFAIK

Reply:
It isn't right? The user of the widget is the one who provide its value.

Reply:
Alright, a bit confusing that it will be `null` when `.controlled` is used, but I'll add that to the dartdocs + docs then.

Reply:
yeah, I agree it can be confusing, but I think dartdoc will be enough , specially because we have two distinct constructors